packages/packagers/js/src/index.js

```diff
@@ -75,6 +78,12 @@ export default (new Packager({
       }));
     }
 
+    if (config.relativeContext) {
+      const relPath = path.relative(bundle.target.distDir, options.projectRoot);
+      contents = contents.replace('$parcel$dirnameReplace', relPath);
+      contents = contents.replace('$parcel$filenameReplace', relPath);
```

Comment:
Probably need to use a regex to replace all instances. I think `replace` of a string only replaces the first one.

Reply:
@devongovett I've added a second call to dirname/filename to my test file and the resulting output is:

```js
var __dirname1 = require("path").join(__dirname, "../test/integration/env-node-replacements");
var __filename = require("path").join(__dirname, "../test/integration/env-node-replacements", "index.js");
const fs = require('fs');
const path = require('path');
module.exports = function() {
    const data = fs.readFileSync(path.join(__dirname1, 'data', 'test.txt'), 'utf8');
    const firstDirnameTest = `${__dirname1}/data`;
    const secondDirnameTest = `${__dirname1}/other-data`;
    const firstFilenameTest = __filename;
    const secondFilenameTest = `${__filename}?query-string=test`;
    return {
        data,
        firstDirnameTest,
        secondDirnameTest,
        firstFilenameTest,
        secondFilenameTest
    };
};
```

So it replaces the calls with only one instance of $parcel$dirnameReplace/$parcel$filenameReplace. In which instances would I have multiple of those?

Reply:
An example:
```js
// index.js
const fs = require("fs"); const path = require("path");
require("./src/other");
const data = fs.readFileSync(path.join(__dirname, "test.txt"), "utf8");
console.log(data);

// src/other.js
const fs = require("fs"); const path = require("path");
const data = fs.readFileSync(path.join(__dirname, "..", "test2.txt"), "utf8");
console.log(data);
```
`__dirname` has to be different in both cases

Reply:
Okay, makes sense 👍  Then I'll adjust the test to incorporate that. How again should I make `$parcel$dirnameReplace` unique per asset? I already forgot 🙈